@AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/simpleRotor/fn_simpleRotor.sqf

```diff
@@ -0,0 +1,172 @@
+//Induced flow handler
+private _velZ                      = velocityModelSpace _heli # 2;
+private _inducedVelocityScalar     = 1.0;
+if (_velZ < -24.384 && _velXY < 12.35) then { 
+    _inducedVelocityScalar = 0.0;
+} else { 
+    _inducedVelocityScalar = 1 - (_velZ / 24.384);
+};
+//Finally, multiply all the scalars above to arrive at the final thrust scalar
+private _rtrThrustScalar           = _bladePitchInducedThrustScalar * _rtrRPMInducedThrustScalar * _airDensityThrustScalar * _airspeedVelocityScalar * _inducedVelocityScalar;
+private _rtrThrust                 = _baseThrust * _rtrThrustScalar;
+private _rtrOmega                  = (2.0 * PI) * ((_rtrDesignRPM * _inputRPM) / 60);
+private _bladeTipVel               = _rtrOmega * _bladeRadius;
+private _rtrArea                   = PI * _bladeRadius^2;
+private _thrustCoef                = if (_rtrOmega == 0) then { 0.0; } else { _rtrThrust / (_dryAirDensity * _rtrArea * _rtrOmega^2 * _bladeRadius^2); };
+_thrustCoef                        = if (_inducedVelocityScalar == 0.0) then { 0.0; } else { _thrustCoef / _inducedVelocityScalar; };
+
+//Calculate the hover induced velocity
+private _rtrInducedVelocity        = sqrt(_rtrThrust / (2 * _dryAirDensity * _rtrArea));
+//Gather the velocities required to determine the actual induced flow velocity using the newton-raphson method
+private _w = _rtrInducedVelocity;
+private _u = if (_w == 0) then { 0.0; } else { _velXY / _rtrInducedVelocity; };
+private _n = if (_w == 0) then { 0.0; } else { _velZ  / _rtrInducedVelocity; };
+private _rtrCorrInducedVelocity    = if (_w == 0) then { 0.0; } else { [_w, _u, _n] call fza_sfmplus_fnc_simpleRotorNewtRaphSolver; };
+_rtrCorrInducedVelocity            = _rtrCorrInducedVelocity * _rtrInducedVelocity;
+//Calculate the required rotor power
+private _rtrPowerScalar            = [_rtrPowerScalarTable, _altitude] call fza_fnc_linearInterp select 1;
+private _rtrPowerReq               = (_rtrThrust * _velZ + _rtrThrust * _rtrCorrInducedVelocity) * _rtrPowerScalar;
+//Calculate the required rotor torque
+private _rtrTorque                 = if (_rtrOmega == 0) then { 0.0; } else { _rtrPowerReq / _rtrOmega; };
+//Calcualte the required engine torque
+private _reqEngTorque              = _rtrTorque / _rtrGearRatio;
+_heli setVariable ["fza_sfmplus_reqEngTorque", _reqEngTorque];
+
+private _axisX = [1.0, 0.0, 0.0];
+private _axisY = [0.0, 1.0, 0.0];
+private _axisZ = [0.0, 0.0, 1.0];
+
+[_heli, _rtrPos, _rtrPos vectorAdd _axisX, _colorRed]   call DRAW_LINE;
+[_heli, _rtrPos, _rtrPos vectorAdd _axisY, _colorGreen] call DRAW_LINE;
+[_heli, _rtrPos, _rtrPos vectorAdd _axisZ, _colorBlue]  call DRAW_LINE;
+
+//Ground Effect
+private _heightAGL    = _rtrHeightAGL  + (ASLToAGL getPosASL _heli # 2);
+private _rtrDiam      = _bladeRadius * 2;
+private _gndEffScalar = (1 - (_heightAGL / _rtrDiam)) * _rtrGndEffModifier;
+_gndEffScalar = [_gndEffScalar, 0.0, 1.0] call BIS_fnc_clamp;
```

Comment:
The ground effect scalar isn't overwritten, it's clamped to ensure it stays within a 0 to 1 value. the closer to the ground the aircraft gets, the closer to 1 it gets (this all depends on the base height of the rotor + altitude above the ground). This then multiplies the thrust and provides the 'cushion effect' of being in ground effect. If you hop in the sim and output that value, it should be at around 0.179 and then trend towards 0 the higher you get, where it reaches 0 at 1 rotor diameter (48 feet, and it's actually a little less because the rotor is 11 ft above the ground) and then you get no additional lift. it has to stay between 0 and 1 or else it becomes infinitely powerful or can go negative, which is equally bad.